src/repositories/ResellerRepository.ts

```diff
@@ -0,0 +1,20 @@
+import { ResellerStorage } from "../storage/ResellerStorage";
+import { Reseller } from "../types/Reseller";
+
+interface IResellerRepository {
+  getReseller(apiKey: string): Reseller | null;
+}
+
+export class ResellerRepository implements IResellerRepository {
+  private storage = new ResellerStorage();
+
+  public getReseller(apiKey: string): Reseller {
+    const reseller = this.storage.get(apiKey);
+
+    if (reseller === null) {
+      throw new Error("Invalid reseller");
```

Comment:
return some http error like bad request, not 500 or handle it later in router or somewhere